### ISSUE TITLE
how to distribute training data to worker nodes?

### ISSUE THREAD

**Issue by OSEN5** [horovod/horovod#1210]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Hi. I have some questions about horovod. 
I just want to know that how partitions of training data are distributed to worker nodes. As I know, horovod doesn't need any parameter server. Then, is there some kind of scheduler solution that send data to workers? Or worker nodes get training data from data store directly?
If there is some document about data distribution in horovod, please let me know.

**Comment by tgaddair** [comment]
Hey @OSEN5, Horovod doesn't prescribe any one way of handling data distribution.  If you look at the examples, we show a couple of different approaches.

In [keras_imagenet_resnet50.py](https://github.com/horovod/horovod/blob/master/examples/keras_imagenet_resnet50.py#L79) we use a random sampling approach.  The [ImageDataGenerator](https://keras.io/preprocessing/image/) provides a random batch at each step, and we divide the total number of steps each epoch by the number of workers to ensure that each worker only processes 1/N of the data.

A more sophisticated approach shown in [keras_spark_rossmann.py](https://github.com/horovod/horovod/blob/master/examples/keras_spark_rossmann.py#L453) is to use [Petastorm](https://github.com/uber/petastorm), which is a library developed by Uber specifically for partitioning data in distributed deep learning jobs.  The only caveat is that Petastorm requires that your data is in the Parquet format.

You can also use TensorFlow datasets, and partition the data with their [shard](https://www.tensorflow.org/versions/r1.12/api_docs/python/tf/data/Dataset#shard) functionality.

So, there are a number of options depending on how your data is formatted and whether you're okay with random sampling (which is easier to configure) or require partitioning.
